h2o-docs/src/product/data-munging/target-encoding.rst

```diff
@@ -146,136 +216,152 @@ Now we will perform target encoding on ``addr_state`` to see if this representat
 
-    head(ext_train[c("addr_state", "fold", "TargetEncode_addr_state")])
+Apply the target encoding to our training and testing data. 
 
-::
+**Apply Target Encoding to Training Dataset** 
 
-    ##   addr_state fold TargetEncode_addr_state
-    ## 1         AK    0               0.1212239
-    ## 2         AK    0               0.1212239
-    ## 3         AK    0               0.1212239
-    ## 4         AK    0               0.1212239
-    ## 5         AK    0               0.1212239
-    ## 6         AK    0               0.1212239
+For our training data, we will use the parameters:
 
-For our testing data, we will use the parameters:
+-  ``holdout_type="kfold"``
+-  ``blended_avg=TRUE``
+-  ``noise=0.2``
 
--  ``holdout_type``: "None"
--  ``blended_avg``: False
--  ``noise_level``: 0
+.. example-code::
+   .. code-block:: r
 
-We do not need to apply any of the overfitting prevention techniques since our target encoding map was created on the training data, not the testing data.
+    ext_train <- h2o.target_encode_transform(train, x = list("addr_state"), y = response, 
+                                             target_encode_map = te_map, holdout_type = "kfold",
+                                             fold_column="fold", blended_avg = TRUE, 
+                                             inflection_point=3, smoothing=1, seed = 1234)
 
-.. code:: r
+   .. code-block:: python
 
-    ext_test <- h2o.target_encode_apply(test, x = list("addr_state"), y = response,
-                                        target_encode_map = te_map, holdout_type = "None",
-                                        fold_column = "fold",
-                                        blended_avg = FALSE, noise_level = 0)
+    ext_train = TargetEncoder(x=predictor, 
+                              y=response,
+                              fold_column="fold",
+                              blended_avg=True, 
+                              inflection_point=3,
+                              smoothing=1, 
+                              seed=1234)
+    encoded_train = te_map.transform(frame=train, holdout_type="kfold", seed=1234)
 
-    head(ext_test[c("addr_state", "TargetEncode_addr_state")])
+**Apply Target Encoding to Testing Dataset**
 
-::
+We do not need to apply any of the overfitting prevention techniques because our target encoding map was created on the training data, not the testing data.
+
+-  ``holdout_type="none"``
+-  ``blended_avg=FALSE``
+-  ``noise=0`` 
+
+.. example-code::
+   .. code-block:: r
+
+    ext_test <- h2o.target_encode_transform(test, x = list("addr_state"), y = response,
+                                            target_encode_map = te_map, holdout_type = "none",
+                                            fold_column = "fold", noise = 0,
+                                            blended_avg = FALSE, seed=1234)
+
+   .. code-block:: python
+
+    ext_test = TargetEncoder(x=predictor, y=response, target_encode_map=te_map,
```

Comment:
@angela0xdata We should not pass `target_encode_map=te_map` and `holdout_type="none"` to TargetEncoder class. TargetEncoder creates `te_map` and holdout_type is needed only during application of the encoding map when we call `.transform()`

Reply:
Thanks for the test. It's helpful (though there are some references to Titanic in it), and I made  a small change, but I'm not able to run that test locally. 
When running the doc example, the following still results in an error.

```
te_map = TargetEncoder(x=predictor, y=response, fold_column="fold", blended_avg= True, inflection_point = 3, smoothing = 1)
te_map.fit(train)
encoded_train = te_map.transform(frame=train, holdout_type="kfold", seed=1234)
```

Error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/h2o/targetencoder.py", line 108, in transform
    assert self._encodingMap.map_keys['string'] == self._teColumns
AssertionError
```

Reply:
@angela0xdata I see the problem. You should pass columns to encode as a list.  
```
teColumns =[ "addr_state"] <----- list here
targetEncoder = TargetEncoder(x= teColumns, y= targetColumnName, fold_column=foldColumnName, blended_avg= True, inflection_point = 3, smoothing = 1)
```
we might want to improve that by accepting just string to make it more robust.

Reply:
created a Jira for that https://0xdata.atlassian.net/browse/PUBDEV-6432

Reply:
@angela0xdata Actually this is a bug as I already added support for passing argument as string. But did not check in the test that calling `transform` method is working fine.

Reply:
@angela0xdata this PR will fix the error that you have been experiencing https://github.com/h2oai/h2o-3/pull/3440. Or just pass predictor as a list.

Reply:
Thanks! I'm just passing it as a list now.